lib/base.js

```diff
@@ -244,21 +244,17 @@ class ConnectionPool extends EventEmitter {
       }
 
       // prepare pool
-      this.pool = gp.createPool({
-        create: this._poolCreate.bind(this),
-        validate: this._poolValidate.bind(this),
-        destroy: this._poolDestroy.bind(this)
-      }, Object.assign({
-        max: 10,
-        min: 0,
-        acquireTimeoutMillis: this.config.connectionTimeout || this.config.timeout || 15000,
-        evictionRunIntervalMillis: 1000,
-        idleTimeoutMillis: 30000,
-        testOnBorrow: true
-      }, this.config.pool))
-
-      this.pool.on('factoryCreateError', this.emit.bind(this, 'error'))
-      this.pool.on('factoryDestroyError', this.emit.bind(this, 'error'))
+      this.pool = new tarn.Pool(
+        Object.assign({
+          create: this._poolCreate.bind(this),
```

Comment:
This is a good PR just for allowing the override of create/validate/destroy on init...